java/rocksjni/ratelimiterjni.cc

```diff
@@ -12,16 +12,21 @@
 /*
  * Class:     org_rocksdb_RateLimiter
  * Method:    newRateLimiterHandle
- * Signature: (JJI)J
+ * Signature: (JJIBZ)J
  */
 jlong Java_org_rocksdb_RateLimiter_newRateLimiterHandle(
     JNIEnv* env, jclass jclazz, jlong jrate_bytes_per_second,
-    jlong jrefill_period_micros, jint jfairness) {
+    jlong jrefill_period_micros, jint jfairness, jbyte jrate_limiter_mode,
+    jboolean jauto_tune) {
+  auto rate_limiter = rocksdb::RateLimiterModeJni::toCppRateLimiterMode(
+      jrate_limiter_mode);
   auto * sptr_rate_limiter =
       new std::shared_ptr<rocksdb::RateLimiter>(rocksdb::NewGenericRateLimiter(
           static_cast<int64_t>(jrate_bytes_per_second),
           static_cast<int64_t>(jrefill_period_micros),
-          static_cast<int32_t>(jfairness)));
+          static_cast<int32_t>(jfairness),
+          rate_limiter,
+          jrate_limiter_mode));
```

Comment:
also pass `jauto_tune`?

Reply:
Oops! I will get that over shortly just knee-deep in some other changes too

Reply:
Okay I fixed, rebased, and pushed